plugins/search-react/src/components/SearchBar/SearchBar.test.tsx

```diff
@@ -288,76 +285,52 @@ describe('SearchBar', () => {
         ]}
       >
         <SearchContextProvider>
-          <SearchBar />
+          <SearchBar debounceTime={0} />
         </SearchContextProvider>
       </TestApiProvider>,
     );
 
-    const textbox = await screen.findByRole('textbox', { name: 'Search' });
+    const textbox = screen.getByLabelText('Search');
 
     const value = 'value';
 
     await user.type(textbox, value);
 
-    act(() => {
-      jest.advanceTimersByTime(200);
-    });
-
-    await waitFor(() => {
-      expect(textbox).toHaveValue(value);
-    });
+    await waitFor(() => expect(textbox).toHaveValue(value));
 
     expect(analyticsApiMock.getEvents()).toHaveLength(0);
-
-    jest.useRealTimers();
   });
 
-  it('captures analytics events if enabled in app', async () => {
-    jest.useFakeTimers();
-    const user = userEvent.setup({ delay: null });
+  it('Captures analytics events if enabled in app', async () => {
+    const analyticsApiMock = new MockAnalyticsApi();
+
     const types = ['techdocs', 'software-catalog'];
 
     await renderWithEffects(
       <TestApiProvider
         apis={[
+          [configApiRef, configApiMock],
           [searchApiRef, searchApiMock],
           [analyticsApiRef, analyticsApiMock],
-          [
-            configApiRef,
-            new ConfigReader({
-              app: {
-                title: 'Mock title',
-                analytics: {
-                  ga: {
-                    trackingId: 'xyz123',
-                  },
-                },
-              },
-            }),
-          ],
         ]}
       >
         <SearchContextProvider initialState={createInitialState({ types })}>
-          <SearchBar />
+          <SearchBar debounceTime={0} />
         </SearchContextProvider>
       </TestApiProvider>,
     );
 
-    const textbox = await screen.findByRole('textbox', { name: 'Search' });
+    const textbox = screen.getByLabelText('Search');
 
     let value = 'value';
 
     await user.type(textbox, value);
 
     expect(analyticsApiMock.getEvents()).toHaveLength(0);
 
-    act(() => {
-      jest.advanceTimersByTime(200);
-    });
+    await waitFor(() => expect(analyticsApiMock.getEvents()).toHaveLength(1));
```

Comment:
This test was flaky. Sometimes the value state was updated before the analytics events trigger, so here we inverted the checking, waiting for analytics events to be captured first. That way we know for sure that both value and events were updated.

Reply:
Alternatively, you could have one waitFor with two expect's at once in the inner function, to detach yourselves from any sequencing risks

Reply:
Hi @freben. I'm taking over this PR since @camilaibs is not available this week.
I'm not sure if I understood you 100%, if added baca9317a85bcc7a952db169e5293f777961a1a9 with a fix attempt , could you take a look?

Reply:
Yeah that's what I had in mind, one waitFor with several expect inside.

I'm out for vacations now though!